### PR TITLE
Add CYBERREEF protocol to nDPI

### PR DESCRIPTION
**Pull request by ChrisNelson-CyberReef**
We won't merge this, it's a proof-of-concept, but a PR is a good way to review changes.

### CODE REVIEW

File: src/lib/ndpi_content_match.c.inc [hunk 1]
```diff
@@ -1655,7 +1655,7 @@ ndpi_protocol_match host_match[] =
    { "xiaomi.net",                       "Xiaomi", NDPI_PROTOCOL_XIAOMI, NDPI_PROTOCOL_CATEGORY_WEB, NDPI_PROTOCOL_SAFE, NDPI_PROTOCOL_DEFAULT_LEVEL },
    { "mi.com",                           "Xiaomi", NDPI_PROTOCOL_XIAOMI, NDPI_PROTOCOL_CATEGORY_WEB, NDPI_PROTOCOL_SAFE, NDPI_PROTOCOL_DEFAULT_LEVEL },
    { "miui.com",                         "Xiaomi", NDPI_PROTOCOL_XIAOMI, NDPI_PROTOCOL_CATEGORY_WEB, NDPI_PROTOCOL_SAFE, NDPI_PROTOCOL_DEFAULT_LEVEL },
-
+
```
Comment (ChrisNelson-CyberReef) [inline src/lib/ndpi_content_match.c.inc:1658]: If we were going to merge this, I'd ask @DeepakDoddigarla-Cyberreef  to revert this change.

File: ndpi-netfilter/ipt/libxt_ndpi.c [hunk 2]
```diff
@@ -800,9 +800,10 @@ void _init(void)
 			   vs = strstr(buf,"#version");
 			   if(!vs)
 	    			xtables_error(PARAMETER_PROBLEM, "xt_ndpi: #version missing");
-			   if(!strstr(vs+8,NDPI_GIT_RELEASE))
-	    			xtables_error(PARAMETER_PROBLEM, "xt_ndpi: module version %s != %s",
-						vs+8,NDPI_GIT_RELEASE);
+			   // TODO - skipping ndpi version
```
Comment (ChrisNelson-CyberReef) [inline ndpi-netfilter/ipt/libxt_ndpi.c:803]: @DeepakDoddigarla-Cyberreef , why is this change needed?

Comment (DeepakDoddigarla-Cyberreef) [reply ndpi-netfilter/ipt/libxt_ndpi.c:803]: While adding iptables-ndpi rules, it throws me error and commenting this part helped me to add rule. I don't think we need an internal build check (which changes frequently)

Comment (ChrisNelson-CyberReef) [reply ndpi-netfilter/ipt/libxt_ndpi.c:803]: I think we should figure out how to fix this.

**Comment by ChrisNelson-CyberReef** [comment]
This was a proof-of-concept.  We're not going to keep or merge it.  Leaving the branch as a clean example of how to add a protocol based only on domains.

**Pull request closed without merging**